wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/kubernetes/KubernetesComponentToWorkspaceApplier.java

```diff
@@ -122,20 +123,24 @@ private String retrieveContent(
           format(
               "Fetching content of file `%s` specified in `reference` field of component `%s` is not supported. "
                   + "Please provide its content in `referenceContent` field. Cause: %s",
-              recipeComponent.getReference(), recipeComponent.getName(), e.getMessage()),
+              recipeComponent.getReference(),
+              getIdentifiableComponentName(recipeComponent),
+              e.getMessage()),
           e);
     } catch (IOException e) {
       throw new DevfileException(
           format(
               "Error during recipe content retrieval for component '%s' with type '%s': %s",
-              recipeComponent.getName(), recipeComponent.getType(), e.getMessage()),
+              getIdentifiableComponentName(recipeComponent),
+              recipeComponent.getType(),
+              e.getMessage()),
           e);
     }
     if (isNullOrEmpty(recipeFileContent)) {
       throw new DevfileException(
           format(
               "The reference file '%s' defined in component '%s' is empty.",
-              recipeComponent.getReference(), recipeComponent.getName()));
+              recipeComponent.getReference(), getIdentifiableComponentName(recipeComponent)));
```

Comment:
I think it's a case that we missed during the discussion. In case of validation issue, we need to point an invalid component. Maybe to make things clearer it makes sense to make alias mandatory again? WDYT? 
Sorry about such late comment, I did not realize how making alias optional affects Che Server.

Reply:
At this point in time, I think the devfile integrity validator has already validated that each component can be identified either by its identifiable attribute or the alias. So in case of this error the "name" you get can pinpoint the concrete component.